rflx/expression.py

```diff
@@ -886,7 +886,10 @@ class Size(Attribute):
 
 
 class Length(Attribute):
-    pass
+    def z3expr(self) -> z3.ArithRef:
+        if isinstance(self.prefix, Aggregate):
+            return z3.IntVal(8 * len(self.prefix.elements))
```

Comment:
You assume here that the elements of an aggregate are always bytes. This should be always the case for the current code, but that is only ensured in the specification parser. In general aggregates can contain arbitrary expressions.

Reply:
Do we ensure anywhere that all elements of an aggregate have the same size? Or can aggregates have elements with different sizes?

Reply:
No, currently not. I see no possibility to ensure that all elements have the same size. That is also not a property of an aggregate in Ada. It only gets such a property from a concrete type (either by assignment or in a comparison).

Reply:
Now I understand. So what we need to do is to find comparisons including aggregates, get the type of the field the aggregate is compared with and extract the element size of that `Composite`. Then we can check whether the element size times the number of elements of the aggregate matches the length of the field. I implemented that in f50a6c632d79a185655d505503ac01a8ad863c63.